etcdserver/apply.go

```diff
@@ -745,3 +745,10 @@ func compareInt64(a, b int64) int {
 func isGteRange(rangeEnd []byte) bool {
 	return len(rangeEnd) == 1 && rangeEnd[0] == 0
 }
+
+func noSideEffect(r *pb.InternalRaftRequest) bool {
+	if r.Txn != nil {
+		return isTxnReadonly(r.Txn)
```

Comment:
Ah I see. BTW, will the txn feature not support putting value that depends on a result of other range reqs in the same transaction? e.g. x = range(); put(modify(x)) in single txn

If there is no such a plan (I'm not sure it is useful or not), your plan seems to be good.